src/azul/indexer/__init__.py

```diff
@@ -250,13 +258,10 @@ def parse(cls, spec: str) -> 'SimpleSourceSpec':
         ...
         azul.uuids.InvalidUUIDPrefixError: '//foo.edu' is not a valid UUID prefix.
         """
-
-        # FIXME: Move parsing of prefix to SourceSpec
-        #        https://github.com/DataBiosphere/azul/issues/3073
-        name, sep, prefix = spec.rpartition(':')
-        reject(sep == '',
-               'Source specifications must end in a colon followed by an optional UUID prefix')
-        return cls(prefix=Prefix.parse(prefix), name=name)
+        rest, prefix = cls._parse(spec)
+        self = cls(prefix=prefix, name=rest)
+        assert spec == str(self), (spec, str(self))
```

Comment:
```suggestion
        assert spec == str(self), spec
```

Only log the smallest possible amount information that is required to reproduce the failing assertion.

Reply:
https://github.com/DataBiosphere/azul/pull/3463#discussion_r741349991

For the case below I've changed the str(prefix) to always return something different, and added your suggested changes.
```python
Index: src/azul/indexer/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/indexer/__init__.py b/src/azul/indexer/__init__.py
--- a/src/azul/indexer/__init__.py	(revision db1c86e8f613cca5d3271295b57e326f3e20793e)
+++ b/src/azul/indexer/__init__.py	(date 1636131199021)
@@ -165,6 +165,7 @@
         >>> s == str(Prefix.parse(s))
         True
         """
+        return 'a'
         if self.partition is None:
             return self.common
         else:
@@ -258,9 +259,9 @@
         ...
         azul.uuids.InvalidUUIDPrefixError: '//foo.edu' is not a valid UUID prefix.
         """
-        rest, prefix = cls._parse(spec)
-        self = cls(prefix=prefix, name=rest)
-        assert spec == str(self), (spec, str(self))
+        name, prefix = cls._parse(spec)
+        self = cls(prefix=prefix, name=name)
+        assert spec == str(self), spec
         return self
 
     def __str__(self) -> str:
```
Here is the failing assertion with your requested change:
```python 
Error
**********************************************************************
File "/Users/amar/dev/hca/azul/src/azul/indexer/__init__.py", line 243, in parse
Failed example:
    SimpleSourceSpec.parse('https://foo.edu:12') # doctest: +NORMALIZE_WHITESPACE
Exception raised:
    Traceback (most recent call last):
      File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/docrunner.py", line 138, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest parse[0]>", line 1, in <module>
        SimpleSourceSpec.parse('https://foo.edu:12') # doctest: +NORMALIZE_WHITESPACE
      File "/Users/amar/dev/hca/azul/src/azul/indexer/__init__.py", line 264, in parse
        assert spec == str(self), spec
    AssertionError: https://foo.edu:12
```

Here it is with the proposed change:
```python
Failed example:
    SimpleSourceSpec.parse('https://foo.edu:12') # doctest: +NORMALIZE_WHITESPACE
Exception raised:
    Traceback (most recent call last):
      File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/docrunner.py", line 138, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest parse[0]>", line 1, in <module>
        SimpleSourceSpec.parse('https://foo.edu:12') # doctest: +NORMALIZE_WHITESPACE
      File "/Users/amar/dev/hca/azul/src/azul/indexer/__init__.py", line 264, in parse
        assert spec == str(self), (spec, str(self))
    AssertionError: ('https://foo.edu:12', 'https://foo.edu:a')
```

In this case it's more helpful to provide the reader/operator with insight to locals that were used in this assertion. It helps isolate where the pain point came from, rather than them having to start from the top and work down to where the failure comes from. In the suggested change I can see that the actual prefix value is malformed.